.github/workflows/test.yml

```diff
@@ -26,6 +26,25 @@ jobs:
         run: just lint-js
       - run: just test-js
 
+  bench-js:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: taiki-e/install-action@v2
+        with: { tool: just }
+      - uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+      - uses: actions/setup-node@v4
+        with:
+          node-version-file: 'js/.nvmrc'
+          cache: 'npm'
+          cache-dependency-path: 'js/package-lock.json'
+      # currently requires java encoder to be run
+      - name: Bench JS code
+        run: just bench-js
```

Comment:
Because the JS requires Java to be set up and the encoder to be compiled it seemed worth its own job, so as not to slow down the JS test job with java install+compile.